manual/src/cmds/profile-perf.etex

```diff
@@ -0,0 +1,90 @@
+
+The basic \texttt{perf} command is:
+\begin{verbatim}
+        perf record -F 99 --call-graph fp <YOUR_EXECUTABLE>
+\end{verbatim}
+
+The \texttt{-F 99} tells perf to sample at 99Hz, which avoids generating excessive data for longer runs and is unlikely to be in lockstep with other periodic activities. The \texttt{--call-graph fp} instructs perf to use frame pointers to get the call-graph (which is the default anyway) and then whatever OCaml executable you want to profile. This will create a \texttt{perf.data} file in the current directory, alternatively use \texttt{--output} to choose a more descriptive filename.
+
+The \texttt{perf record} command works by copying (a segment) of the call stack at every sample and recording this into a perf.data file. These samples are then processed (using \texttt{perf report} to reconstruct the profiled program's call stack at every sample) offline, after recording has finished.
+
+\texttt{perf record --call-graph} has multiple options for record call graphs:
+\begin{itemize}
+\item Frame Pointers, which is the default
+\item DWARF's Call Frame Information (CFI)
+\item Hardware Last Branch Record (LBR) available on certain Intel CPUs
+\end{itemize}
+
+Perf will use the symbols present in an OCaml executable, so it is useful to understand OCaml's name mangling scheme to map these names to OCaml source locations. Before 5.1, ocamlopt mangles names using the following format "camlModule__identifier_stamp" while 5.1 onwards replaces the double underscore separator with a dot "camlModule.identifier_stamp". Both options are supported by perf.
+
+For example, consider the following program:
+\begin{caml_example*}{verbatim}
+module Compute = struct
+  let rec fib n = 
+    if n = 0 then 0
+    else if n = 1 then 1
+    else fib (n-1) + fib (n-2)
+end
+
+let main () = 
+  let r = Compute.fib 20 in 
+  Printf.printf "fib(20) = %d" r
+
+let _ = main ()
+\end{caml_example*}
+
+This will produce the following names \texttt{camlFib.main_274} for \texttt{main} function and \texttt{camlFib.fib_271} for the \texttt{fib} function in the \texttt{Compute} module when compiled with OCaml 5.2.
+
+\section{s:ocamlperf-printing}{Printing profiling information}
+
+The basic \texttt{perf} reporting command is:
+
+\begin{verbatim}
+        perf report -f --no-children perf.data
+\end{verbatim}
+
+this will display the accumulated call-graphs in an interactive terminal interface where you can navigate the data, select functions and threads for more information. \texttt{--stdio} will output present similar data using a text based report.
+
+Alternatively profiles can be turned into FlameGraphs, a visualization of hierarchical data, created to visualize stack traces of profiled software so that the most frequent code-paths to be identified quickly and accurately. Use the scripts found at Brendan Gregg's \href{https://www.brendangregg.com/flamegraphs.html}{Flame Graphs} web page as follows:
```

Comment:
```suggestion
Alternatively profiles can be turned into FlameGraphs, a visualization of hierarchical data, created to visualize stack traces of profiled software so that the most frequent code-paths can be identified quickly and accurately. Use the scripts \texttt{stackcollapse-perf.pl} and \texttt{flamegraph.pl} found at Brendan Gregg's \href{https://www.brendangregg.com/flamegraphs.html}{Flame Graphs} web page as follows:
```